types/roslib/index.d.ts

```diff
@@ -77,173 +78,258 @@ declare namespace ROSLIB {
+         * @param {function} callback - Function with the following params:
+         * @param {string[]} callback.nodes - Array of node names.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
          */
-        getNodes(callback: (nodes: string[]) => void, failedCallback?: (error: any) => void): void;
+        getNodes(callback: (nodes: string[]) => void, failedCallback?: (error: string) => void): void;
 
         /**
-         * Retrieves list of param names from the ROS Parameter Server.
+         * Retrieve a list of subscribed topics, publishing topics and services of a specific node.
+         * <br>
+         * These are the parameters if failedCallback is <strong>defined</strong>.
          *
-         * @param callback function with params:
-         *  * params - array of param names.
-         * @param failedCallback - the callback function when the ros call failed (optional). Params:
-         *   * error - the error message reported by ROS
+         * @param {string} node - Name of the node.
+         * @param {function} callback - Function with the following params:
+         * @param {string[]} callback.subscriptions - Array of subscribed topic names.
+         * @param {string[]} callback.publications - Array of published topic names.
+         * @param {string[]} callback.services - Array of service names hosted.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
+         *
+         * @also
+         *
+         * Retrieve a list of subscribed topics, publishing topics and services of a specific node.
+         * <br>
+         * These are the parameters if failedCallback is <strong>undefined</strong>.
+         *
+         * @param {string} node - Name of the node.
+         * @param {function} callback - Function with the following params:
+         * @param {Object} callback.result - The result object with the following params:
+         * @param {string[]} callback.result.subscribing - Array of subscribed topic names.
+         * @param {string[]} callback.result.publishing - Array of published topic names.
+         * @param {string[]} callback.result.services - Array of service names hosted.
+         * @param {function} [failedCallback] - The callback function when the service call failed with params:
+         * @param {string} failedCallback.error - The error message reported by ROS.
+         */
+        getNodeDetails(
+            node: string,
+            callback: ((subscriptions: string[], publications: string[], services: string[]) => void) | ((result: { subscribing: string[], publishing: string[], services: string[] }) => void),
+            failedCallback?: (error: string) => void,
+        ): void;
```

Comment:
Better to define `getNodeDetails` with type overloading.

Reply:
@alumag Note that this breaks the tests with the old `tslint.json` file, and thus, I have set `unified-signatures` to be `false`.

Reply:
@alumag Yup I agree with you, I was simply making a note of the change to the `tslint.json` file to ensure that you are aware of it.

I have separated the two definitions via type overloading in commit 6018f629dab248db572ebd58ce6e32bffe37dcca and I have set `unified-signatures` to `false` in commit 452f2d2cf98d6e644fe3673c127fd07f82c4576a.